crates/net/eth-wire/src/p2pstream.rs

```diff
@@ -613,19 +613,31 @@ where
     /// Returns `Poll::Ready(Ok(()))` when no buffered items remain.
     fn poll_flush(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
         let mut this = self.project();
-        loop {
-            match ready!(this.inner.as_mut().poll_flush(cx)) {
-                Err(err) => return Poll::Ready(Err(err.into())),
-                Ok(()) => {
+        let poll_res = loop {
+            // P2PStream wraps `reth_ecies::stream::ECIESStream` in node. `ECIESStream`
+            // internally wraps the network connection in `tokio_util::codec::Framed`.
+            //
+            // Call to `Framed::poll_ready` will flush internally, if limit for buffered messages
+            // reached. `Framed::poll_ready`, only returns `Poll::Pending` or `Poll::Ready(Err(e))`
+            // if it had to call flush - no need to explicitly flush the batch again before
+            // propagating pending or error.
```

Comment:
these are specialized docs for framed but this type is generic over any sink type, please rephrase to be general